app/models/job_run.rb

```diff
@@ -3,15 +3,33 @@
 class JobRun < ApplicationRecord
   belongs_to :batch_context
   validates :job_type, presence: true
+  validates :state, presence: true
   after_initialize :default_enums
   after_create :enqueue!
-  after_update :send_notification, if: -> { saved_change_to_output_location? }
 
   enum job_type: {
     'discovery_report' => 0,
     'preassembly' => 1
   }
 
+  state_machine initial: :waiting do
+    event :started do
+      transition waiting: :running
+    end
+
+    event :errored do
+      transition running: :error
+    end
+
+    event :completed do
+      transition running: :complete
+    end
+
+    after_transition do |job_run, transition|
+      job_run.send_notification if transition.from_name == :running
+    end
+  end
+
```

Comment:
use the state transition away from running to send the email notification instead of depending on the output location field being changed

Reply:
I would have assumed it was sending email if something was saved to the output location, but whatevs - I like the state machine better.